subsys/bluetooth/mesh/gen_loc_srv.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "bluetooth/mesh/gen_loc.h"
+#include <string.h>
+#include <bluetooth/mesh/gen_loc_srv.h>
+#include <bluetooth/mesh/gen_dtt_srv.h>
+#include "model_utils.h"
+#include "gen_loc_internal.h"
+
+#define LOC_GLOBAL_DEFAULT                                                     \
+	{                                                                      \
+		.latitude = 0.0, .longitude = 0.0,                             \
+		.altitude = GEN_LOC_ALTITUDE_UNKNOWN,                          \
+	}
+
+#define LOC_LOCAL_DEFAULT                                                      \
+	{                                                                      \
+		.north = 0, .east = 0, .altitude = GEN_LOC_ALTITUDE_UNKNOWN,   \
+		.floor_number = GEN_LOC_FLOOR_NUMBER_UNKNOWN,                  \
+		.is_mobile = false, .time_delta = 0, .precision_mm = 4096000,  \
+	}
+
+static bool pub_in_progress(struct gen_loc_srv *srv)
+{
+	return (k_delayed_work_remaining_get(&srv->pub.timer) > 0) ||
```

Comment:
Why is there special consideration for this check during publication compared to other models?

Reply:
This is the only model that actually publishes different state periodically, so to avoid the set-related publish to replace periodic publishing of the other state, we check whether we'll be interrupting anything before doing the publish.

Reply:
Hmm, I see. Then, could it be better to prioritise the publication resulting from Set over periodic? Since cancelling Set response will trigger retransmissions, and if periodic publication is frequent enough, it will lead to difficulties in completing a Set transaction.

Reply:
I'm not quite sure I follow. Why would frequent periodic publishing impact the set-transaction? We will send the response regardless of publishing activity, as the checks on line 80 and line 146 only affect publishing to the publish address, ie the arrow going to "Publish Group" in Mesh core spec 1.0.1 figure 3.28.

Reply:
What I meant was this: Assuming model's periodic publish period is 1 minute (or longer), and if the user/tester expects a model to publish a message as a result of the state change after receiving acknowledged Set message, then because of this check, such state change will be published only when the publish interval expires causing delayed publication of the status. 

Also, what happens when the user calls publish APIs asynchronously without being aware of publish period settings?

Reply:
Right, I see, that's a good point. The timer will reset if we do a publish, there's no way around this with the current API, I'm afraid.

Reply:
Ok. I believe resetting of a timer is fine, as long as this behaviour is documented. Later this can improved.